core/src/main/java/com/gamification/core/entities/User.java

```diff
@@ -1,23 +1,67 @@
 package com.gamification.core.entities;
 
+import com.gamification.core.enumeration.Role;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
+import org.springframework.security.core.CredentialsContainer;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
 
 import javax.persistence.*;
-
+import java.util.Collection;
+import java.util.List;
 
 
 @Table(name = "USERS")
 @Entity
-@Getter@Setter@NoArgsConstructor
-public class User extends BaseEntity{
+@Getter
+@Setter
+@NoArgsConstructor
+public class User extends BaseEntity implements UserDetails, CredentialsContainer {
 
     private String username;
     private String password;
-    private Boolean isActive;
-
     @ManyToOne
-    @JoinColumn(name = "PERSON_ID",foreignKey = @ForeignKey(name = "FK_PERSON_USER"))
+    @JoinColumn(name = "PERSON_ID", foreignKey = @ForeignKey(name = "FK_PERSON_USER"))
     private Person person;
+
+    @Enumerated(EnumType.STRING)
+    private Role role;
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return List.of(new SimpleGrantedAuthority(role.name()));
+    }
+
+    @Override
+    public String getUsername() {
+        return username;
+    }
+
+    @Override
+    public boolean isAccountNonExpired() {
```

Comment:
please define field for these method to return them instead of returning true for every scenario.